City-of-Helsinki/drupal-helfi-kasvatus-koulutus
# UHF-8615 groups

# [UHF-8615](https://helsinkisolutionoffice.atlassian.net/browse/UHF-8615)

Updated groups module to be D10 ready.

## How to install

In order to get this working, you must have flexible_permissions module installed and enabled beforehand.

* Make sure your instance is up and running on correct branch.
  * `git checkout dev`
  * `make fresh`
  * `git checkout UHF-8615_groups`
  * run `composer install`
  * run ` drush cr`
  * run `drush updb`
  * run `drush cr`
  * run `drush cim`
  * run `drush cr`


## How to test
FYI: UID1 account cannot see groups other than own.

- Go to testing and select users related to groups, use `drush uli` to log in to testing environment with those users
- Login with the same users to local dev environment
- Cross reference everything related to groups & group content
  - Check that users can see the same groups on local and testing
  - Check that users can edit the same content
Also test: 
  - Membership handling
  - Permissions
  - Translations
  - Menu handling


[UHF-8615]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-8615?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
